### DISCUSSION TITLE
使用内网穿透部署时 site.url 错误

### DISCUSSION THREAD

**Discussion by purainity** [halo-dev/halo#7118] [Q&A]
在本地通过 docker 部署 halo，使用 cloudflare tunnel 内网穿透实现外部访问。但是通过外部域名访问时的 site.url 和 og:url 仍然为内网 IP+端口，而不是自动识别。
![Screenshot_2024-12-06-19-33-32-509_mark via](https://github.com/user-attachments/assets/45c6aee7-57da-4ca9-94e8-8ddf3ccdaabd)

**Comment by ruibaby** [comment]
你需要正确设置外部访问地址，Halo 不会把访问的 Host 当做网站地址。

https://docs.halo.run/getting-started/install/config#halo-%E7%8B%AC%E6%9C%89%E9%85%8D%E7%BD%AE

**Reply by purainity** [reply]
在内网直接用 IP+端口访问更方便，在外网就用域名访问。请问为什么不加一个自动获取 site.url 的功能？

**Reply by ruibaby** [reply]
没有理解你的意思，这个地址设置仅仅只是确定一个网站地址，这可能会在发送邮件（比如评论通知中，要在邮件中构建文章地址）等地方使用，系统要发送邮件的时候怎么根据请求拿到一个 url？

除了这些要明确获取一个固定地址的地方，不管你使用内网还是外网访问都可以正常使用。所以我不明白你的问题到底在哪里。
